src/libraries/System.Collections.Immutable/tests/IndexOfTests.cs

```diff
@@ -44,6 +44,8 @@ public static class IndexOfTests
             if (factory(list) is IList)
             {
                 Assert.Equal(-1, ((IList)factory(list)).IndexOf(6));
```

Comment:
This is testing `ImmutableList<T>`. While the additions are nice to have, could you also add a test for the updated `ImmutableArray` methods?

Reply:
Are these testing the IList IndexOf method and are they accounting for mistyped inputs?

Reply:
If I understand your question correctly, yes. I put them in before making the fix and the test was failing until the fix went in.  They are testing for null and "a" on IList<int> collections.

Reply:
Ah I see, so presumably one of the `factory` parameters being passed by the test converts the `ImmutableList` to an `ImmutableArray`?

Reply:
The factory parameter creates either an ImmutableArray or ImmutableList depending on what is being tested.